Adding Gengo Text to YAML.

AntiFish03 opened the pull request:
Added a Jasmine suite to test Object -> Gengo text conversion, and  Gengo Text -> Object
Using Bootstrap 4 to style.

Commit:
Adding Gengo Text to YAML.
Added a Jasmine suite to test Object -> Gengo text conversion, and  Gengo Text -> Object
Using Bootstrap 4 to style.

Commit:
Forgot to wire up the copy to clipboard and disable the send to gengo button unless the converted is Gengo Text

leighmcculloch commented:
@AntiFish03: What's the main change you're proposing to be included? It's not clear from reading the commit messages or looking at the code as I can see multiple major changes:
- Adding specs for existing logic.
- Introducing the use of Bootstrap to replace the plain JavaScript.
- Significant changes to classes and DOM structure.
- Style changes (changing double-quotes to single-quotes).

👍  for adding the tests, but could you outline what the motivation for the other changes are? Ideally each change would be independent and in separate PRs also.

AntiFish03 commented:
@leighmcculloch,  a few things.   The company I work for is in the process of a lot of i18n upgrades, and we needed a tool quickly to go from yaml to Gengo text.   Your utility fit that bill but we also needed to reverse the process Gengo text to yaml which still needed to occur. 

So,  there needed to be a few changes.  The first change was the yaml js vendor script so it could go from yaml to native js object and native js object back to yaml.   Using that same methodology I modified and abstracted your creation of a gengo adapter into a reusable object that also goes gengo text to native and native to gengo text.   This allowed me to simplify  the index.js file to UI interactions only.  The double to single quotes was me getting rid of the jshint warnings in my development workflow.  The changes to the UI were to facilitate the use of both directions. Yaml to gengo and gengo back to yaml.   I added boot strap because I use it a great deal and is very powerful for easy to use form interfaces and there was a considerable amount of change for the types of interaction a user could make. 

I can break this into many prs if I have to but that will have to occur on my personal time not the time at work.  I urge you to go over to go try it on my github page  https://antifish03.github.io/gengo-yaml/ and see what you think.

leighmcculloch commented:
@AntiFish03 I understand. It sounds like it snowballed as you went 😄. I like the functionality you added and the tests, but I'm not wanting to dramatically increase the complexity of this project by adding dependencies like bootstrap and jasmine, or change the existing style of the project. It's not clear where the _yaml.js file has come from (sorry if it's obvious and I'm missing it) and so the licensing of it is not clear. I'd be up for maybe doing some of these things step-by-step, but it's very difficult to review multi-faceted commits, and to weigh each individual facet/change on it's own merits, with it's own pro's and con's.

I'm going to close this PR because of this ^, but your changes are basically a complete rewrite and would serve well as an independent project / fork. If you feel passionate about that I want to encourage you to carry this on in that way, as a fork. :+1:

AntiFish03 commented:
@leighmcculloch  My version will stay where it is, I have need of it. And I understand your concerns, and your reasoning... I needed something slightly different from yours so I used yours as a starting point. Feel free in the future to snag any of the changes you might want to add to yours. I don't mind in the slightest.

Commit:
Adding google analytics page views

Commit:
Using GA async method, adding click event tracking

Commit:
Only trigger ga event when the cta is clicked. not on method call

Commit:
Fixing issue with duplicate GA tracking ids... Removing original
